src/Sentry/SentryMonitorOptions.cs

```diff
@@ -0,0 +1,224 @@
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="crontab"></param>
+    public void Interval(string crontab)
+    {
+        if (_type is not SentryMonitorScheduleType.None)
+        {
+            throw new ArgumentException("You tried to set the interval twice. The Check-Ins interval is supposed to be set only once.");
+        }
+
+        if (!CrontabValidation().IsMatch(crontab))
+        {
+            throw new ArgumentException("The provided crontab does not match the expected format of '* * * * *' " +
+                                        "translating to 'minute', 'hour', 'day of the month', 'month', and 'day of the week'.");
+        }
+
+        _type = SentryMonitorScheduleType.Crontab;
+        _crontab = crontab;
+    }
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="interval"></param>
+    /// <param name="unit"></param>
+    public void Interval(int interval, SentryMonitorInterval unit)
+    {
+        if (_type is not SentryMonitorScheduleType.None)
+        {
+            throw new ArgumentException("You tried to set the interval twice. The Check-Ins interval is supposed to be set only once.");
+        }
+
+        _type = SentryMonitorScheduleType.Interval;
+        _interval = interval;
+        _unit = unit;
+    }
+
+    /// <summary>
+    /// The allowed margin of minutes after the expected check-in time that the monitor will not be considered missed for.
+    /// </summary>
+    public TimeSpan? CheckInMargin { get; set; }
+
+    /// <summary>
+    /// The allowed duration in minutes that the monitor may be in_progress for before being considered failed due to timeout.
+    /// </summary>
+    public TimeSpan? MaxRuntime { get; set; }
+
+    private int? _failureIssueThreshold;
+
+    /// <summary>
+    /// The number of consecutive failed check-ins it takes before an issue is created.
+    /// </summary>
+    public int? FailureIssueThreshold
+    {
+        get => _failureIssueThreshold;
+        set
+        {
+            if (value <= 0)
+            {
+                throw new ArgumentException("FailureIssueThreshold has to be set to a number greater than 0.");
+            }
+            _failureIssueThreshold = value;
+        }
+    }
+
+    private int? _recoveryThreshold;
+
+    /// <summary>
+    /// The number of consecutive OK check-ins it takes before an issue is resolved.
+    /// </summary>
+    public int? RecoveryThreshold
+    {
+        get => _recoveryThreshold;
+        set
+        {
+            if (value <= 0)
+            {
+                throw new ArgumentException("RecoveryThreshold has to be set to a number greater than 0.");
+            }
+            _recoveryThreshold = value;
+        }
+    }
+
+    /// <summary>
+    /// A tz database string representing the timezone which the monitor's execution schedule is in (i.e. "America/Los_Angeles").
+    /// </summary>
+    public string? Timezone { get; set; }
```

Comment:
IIRC I raised this before but it's `TimeZone`: not Timezone: https://learn.microsoft.com/en-us/dotnet/api/system.timezone?view=net-8.0